lua/nvim-tree/lib.lua

```diff
@@ -213,8 +213,85 @@ function M.set_index_and_redraw(fname)
   end
 end
 
+---Get user to pick a window. Selectable windows are all windows in the current
+---tabpage that aren't NvimTree.
+---@return integer|nil -- If a valid window was picked, return its id. If an
+--invalid window was picked, return nil. If there are no selectable windows,
+--return -1.
+function M.pick_window()
+  local tabpage = api.nvim_get_current_tabpage()
+  local win_ids = api.nvim_tabpage_list_wins(tabpage)
+  local tree_winid = view.View.tabpages[tabpage]
+  local selectable = {}
+
+  for _, id in ipairs(win_ids) do
+    if id ~= tree_winid then
+      table.insert(selectable, id)
+    end
+  end
+
+  -- If there are no selectable windows: return. If there's only 1, return it without picking.
+  if #selectable == 0 then return -1 end
+  if #selectable == 1 then return selectable[1] end
+
+  local chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+  if vim.g.nvim_tree_window_picker_chars then
+    chars = tostring(vim.g.nvim_tree_window_picker_chars):upper()
+  end
+
+  local i = 1
+  local win_opts = {}
+  local win_map = {}
+  local laststatus = vim.o.laststatus
+  vim.o.laststatus = 2
+
+  -- Setup UI
+  for _, id in ipairs(selectable) do
+    local char = chars:sub(i, i)
+    local _, statusline = pcall(api.nvim_win_get_option, id, "statusline")
+    local _, winhl = pcall(api.nvim_win_get_option, id, "winhl")
+
+    win_opts[id] = {
+      statusline = statusline or "",
+      winhl = winhl or ""
+    }
+    win_map[char] = id
+
+    api.nvim_win_set_option(id, "statusline", "%=" .. char .. "%=")
+    api.nvim_win_set_option(
+      id, "winhl", "StatusLine:NvimTreeWindowPicker,StatusLineNC:NvimTreeWindowPicker")
```

Comment:
this looks strangely formatted, could you just put the end parens in the next line ?